yt_dlp/extractor/servus.py

```diff
@@ -1,148 +1,304 @@
         return {
             'id': video_id,
-            'title': title,
-            'alt_title': alt_title,
-            'description': description,
-            'thumbnail': thumbnail,
-            'duration': duration,
-            'timestamp': unified_timestamp(video.get('lastPublished')),
-            'series': series,
-            'season': season,
-            'season_number': season_number,
-            'episode': episode,
-            'episode_number': episode_number,
+            'title': info.get('title'),
+            'description': info.get('description'),
+            'thumbnail': info.get('poster'),
+            'duration': None if is_live else info.get('duration'),
+            'timestamp': parse_iso8601(info.get('currentSunrise')),
+            'is_live': is_live,
+            'categories': [info['label']] if info.get('label') else [],
+            'age_limit': int(
+                self._search_regex(r'(?:^|\s)(\d\d?)(?:\s|$)', info.get('maturityRating', '0'),
+                                   'age_limit', default='0')),
             'formats': formats,
+            'subtitles': subtitles,
         }
+
+    def _live_stream_from_schedule(self, schedule):
+        live_url = self._LIVE_URLS['AT']
+        for item in sorted(schedule, key=lambda x: x.get('is_live', False), reverse=True):
+            is_live = item.get('is_live', False)
+            video_url = self._LIVE_URLS.get(self.country_code, live_url) if is_live else None
+            return self._entry_by_id(item['aa_id'].lower(), video_url=video_url, is_live=is_live)
+
+    def _paged_playlist_by_query(self, query_type, query_id, extra_query=(), page_size=20, ie=None):
+        query = {query_type: query_id, 'geo_override': self.country_code,
+                 'post_type': 'media_asset', 'filter_playability': 'true', 'per_page': page_size}
+        assert 'per_page' not in extra_query
+        query.update(extra_query)
+
+        def fetch_page(page_number):
+            query.update({'page': page_number + 1})
+            info = self._download_json(
+                self._QUERY_API_URL,
+                query=query,
+                video_id=f'{query_type}-{query_id}',
+                note=f"Downloading entries "
+                     f"{page_number * page_size + 1}-{(page_number + 1) * page_size}")
+
+            for item in info['posts']:
+                if not traverse_obj(item, ('stv_duration', 'raw')):
+                    continue
+                video_id, title, url = itemgetter('slug', 'stv_short_title', 'link')(item)
+                yield self.url_result(
+                    url, ie=ie or self.ie_key(), video_id=video_id, video_title=title)
+
+        return OnDemandPagedList(fetch_page, page_size)
+
+    @staticmethod
+    def _page_id(json_obj):
+        for key, value in traverse_obj(json_obj, ('source', 'data'), default={}).items():
+            if isinstance(value, dict) and 'id' in value:
+                return value['id']
+        return None
```

Comment:
this shouldnt even be a function
```suggestion
    def _page_id(json_obj):
        return traverse_obj(json_obj, ('source', 'data', ..., 'id'))
```